### PR TITLE
Add eslint, prettier, ls-lint configs

### PR DESCRIPTION
**Pull request by kabaros**
This PR ensures our linting and formatting rules are applied properly through d2-style.

- add `eslint`, `prettier`, `ls-lint` configs
   - initially, we added d2-style but didn't add the config template files, i.e. ` yarn d2-style add eslint react`
- run `yarn format` so that all files are formatted properly.
- Add git commit hook to check the rules on: `yarn d2-style add git-hook pre-commit "yarn d2-style apply && git add -u"`
- run a basic workflow in CI - added using `yarn d2-style add github workflow-app`

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { config } = require('@dhis2/cli-style')
+
+module.exports = {
```
Comment (Birkbjo) [inline .eslintrc.js:4]: I never added my `eslintrc`... This is what I ended up with

const { config } = require("@dhis2/cli-style");
//import { config } from "@dhis2/cli-style";

module.exports = {
    extends: [
        config.eslintReact,
        "plugin:@typescript-eslint/recommended",
        "plugin:import/typescript",
    ],
    parser: "@typescript-eslint/parser",
    plugins: ["@typescript-eslint"],
    root: true,
    rules: {
        "@typescript-eslint/no-var-requires": 0,
        "import/extensions": 0,
        "react/prop-types": 0, // we are using types instead
    },
};

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    extends: [config.eslintReact, 'plugin:@typescript-eslint/recommended'],
+    settings: {
+        'import/resolver': {
+            node: {
+                extensions: ['.js', '.jsx', '.ts', '.tsx'],
+            },
+        },
+    },
+    rules: {
+        'import/extensions': 'off',
```
Comment (Mohammer5) [inline .eslintrc.js:14]: I think we decided to have these (not sure why, though)

File: .ls-lint.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+ls:
+    src:
+        .dir: PascalCase | camelCase
```
[the remark this answers is not in the document]
Comment (Mohammer5) [reply .ls-lint.yml:3]: Is there a way to make these more precise? The idea was that each folder reflects what's inside. That could be a component (PascalCase) or a function/an entire module (camelCase)

**Review by Mohammer5 — APPROVED** [review]
Left two comments, I don't care too much about them, I'd be OK with the current state of the PR (no file extensions & simply Pascal- or camelCase)

File: .eslintrc.js (hunk 2 above)
Comment (Birkbjo) [reply .eslintrc.js:14]: Import extensions is not compatible with typescript as far as I know. You would also need to enable `allowImportingTsExtensions`.

**Pull request merged**